File: todo-ui/src/services/TodoService.js
import axios from "axios";

const BASE_REST_API_URL = 'http://localhost:8080/api/todos';

export function getAllTodos(){
    return axios.get(BASE_REST_API_URL);
}

export function addTodo(todo){
    return axios.post(BASE_REST_API_URL, todo)
}

export function getTodo(id){
    return axios.get(BASE_REST_API_URL + '/' + id);
}

export function updateTodo(id, todo){
    return axios.put(BASE_REST_API_URL + '/' + id, todo);
}

export function deleteTodo(id){
    return axios.delete(BASE_REST_API_URL + '/'+ id);
}

export function completeDBTodo(id){
    return axios.patch(BASE_REST_API_URL + '/' + id + '/complete');
}

export function inCompleteDBTodo(id){
    return axios.patch(BASE_REST_API_URL + '/' + id + '/in-complete');
}

File: todo-management/src/main/java/com/example/todo/service/TodoService.java
package com.example.todo.service;

import com.example.todo.dto.TodoDto;

import java.util.List;

public interface TodoService {

    TodoDto addTodo(TodoDto todoDto);

    TodoDto getTodo(Long id);

    List<TodoDto> getAllTodos();

    TodoDto updateTodo(Long id, TodoDto todoDto);

    void deleteTodo(Long id);

    TodoDto completeTodo(Long id);

    TodoDto inCompleteTodo(Long id);

}
